iotdb/src/main/java/org/apache/iotdb/db/conf/adapter/IoTDBConfigDynamicAdapter.java

```diff
@@ -0,0 +1,226 @@
+import org.apache.iotdb.db.engine.StorageEngine;
+import org.apache.iotdb.db.exception.ConfigAdjusterException;
+import org.apache.iotdb.db.metadata.MManager;
+import org.apache.iotdb.db.rescon.PrimitiveArrayPool;
+
+public class IoTDBConfigDynamicAdapter implements IDynamicAdapter {
+
+
+  private static final IoTDBConfig CONFIG = IoTDBDescriptor.getInstance().getConfig();
+
+  // static parameter section
+
+  private static final double FLUSH_THRESHOLD = 0.2;
+
+  /**
+   * Maximum amount of memory allocated for write process.
+   */
+  private static final long ALLOCATE_MEMORY_FOR_WRITE = CONFIG.getAllocateMemoryForWrite();
+
+  /**
+   * Metadata size of per timeseries, the default value is 2KB.
+   */
+  private static final long TIMESERIES_METADATA_SIZE_IN_BYTE = 2L * 1024;
+
+  /**
+   * Metadata size of per chunk, the default value is 1.5 KB.
+   */
+  private static final long CHUNK_METADATA_SIZE_IN_BYTE = 1536L;
+
+  /**
+   * Average queue length in memtable pool
+   */
+  static final int MEM_TABLE_AVERAGE_QUEUE_LEN = 5;
+
+  // static memory section
+
+  /**
+   * Static memory, includes all timeseries metadata, which equals to TIMESERIES_METADATA_SIZE_IN_BYTE *
+   * totalTimeseriesNum, the unit is byte
+   */
+  private long staticMemory;
+
+  private int totalTimeseries;
+
+  // MemTable section
+
+  private int maxMemTableNum = MEM_TABLE_AVERAGE_QUEUE_LEN;
+
+  private int currentMemTableSize;
+
+  // Adapter section
+
+  private boolean initialized = false;
+
+  @Override
+  public synchronized boolean tryToAdaptParameters() {
+    boolean shouldAdjust = true;
+    int memtableSizeInByte = calcMemTableSize();
+    int memTableSizeFloorThreshold = getMemTableSizeFloorThreshold();
+    boolean shouldClose = false;
+    long tsFileSize = CONFIG.getTsFileSizeThreshold();
+    if (memtableSizeInByte < memTableSizeFloorThreshold) {
+      shouldClose = true;
+      tsFileSize = calcTsFileSize(memTableSizeFloorThreshold);
+      memtableSizeInByte = (int) tsFileSize;
+      if (tsFileSize < memTableSizeFloorThreshold) {
+        shouldAdjust = false;
+      }
+    } else if (memtableSizeInByte > tsFileSize) {
+      memtableSizeInByte = (int) tsFileSize;
+    }
+
+    if (shouldAdjust) {
+      CONFIG.setMaxMemtableNumber(maxMemTableNum);
+      CONFIG.setTsFileSizeThreshold(tsFileSize);
+      CONFIG.setMemtableSizeThreshold(memtableSizeInByte);
+      if(initialized) {
+        if (shouldClose) {
+          StorageEngine.getInstance().asyncFlushAndSealAllFiles();
+        } else if (memtableSizeInByte < currentMemTableSize
+            && currentMemTableSize - memtableSizeInByte > currentMemTableSize * FLUSH_THRESHOLD) {
+          StorageEngine.getInstance().asyncFlushAllProcessor();
+        }
+      }
+      currentMemTableSize = memtableSizeInByte;
+    }
+    if (!initialized) {
+      CONFIG.setMaxMemtableNumber(maxMemTableNum);
+      return true;
+    }
+    return shouldAdjust;
```

Comment:
return find result

Reply:
I don't get the point. Please be more specific.